torchtune/data/_prompt_templates.py

```diff
@@ -161,12 +163,24 @@ def __call__(
             The formatted list of messages
         """
         formatted_dialogue = []
-        for message in messages:
-            content = (
-                [{"type": "text", "content": self.template[message.role][0]}]
-                + message.content
-                + [{"type": "text", "content": self.template[message.role][1]}]
-            )
+        for index, message in enumerate(messages):
```

Comment:
Was this a bug from before?

Reply:
kind of, see this discussion: https://github.com/pytorch/torchtune/pull/1019/files#r1621540801. it was also called out in the Qwen2 PR (#1143): 
>The official implemtation torchtune.data._chat_formats.ChatMLFormat will add an extra <|IM_END|> tag at the end of the last assistant message in generation (see [this discussion](https://github.com/pytorch/torchtune/pull/1019/files#r1621540801)), so we override the tokenize_message() method to fix this problem (will not add <|IM_END|> and EOS token if messages[-1].role == 'assistant' and not messages[-1].content)